### PR TITLE
docs: public key crypto specs

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1440

### CODE REVIEW

File: specs/src/specs/data_structures.md [hunk 1]
```diff
@@ -172,24 +167,18 @@ Unless otherwise indicated explicitly, objects are first [serialized](#serializa
...
-
-`v` represents the parity of the `Y` component of the point, `0` for even and `1` for odd. The `X` component of the point is assumed to always be low, since [the possibility of it being high is negligible](https://bitcoin.stackexchange.com/a/38909).
-
-Putting it all together, the encoding for signatures is:
+[Signatures](#signature) are represented as the `r` and `s` (each 32 bytes) values of the signature. `r` and `s` take on their usual meaning (see: [SEC 1, 4.1.3 Signing Operation](https://www.secg.org/sec1-v2.pdf)). The encoding for signatures is:
```
Comment (rootulp) [inline specs/src/specs/data_structures.md:174]: @adlerjohn I'd appreciate your review on this section

File: specs/src/specs/data_structures.md [hunk 2]
```diff
@@ -112,11 +111,7 @@ Abstraction over transaction fees.
 
 ### Address
 
-Address is a [type alias](#type-aliases).
-
-Addresses are the [hash](#hashing) [digest](#hashdigest) of the [public key](https://docs.cosmos.network/v0.46/basics/accounts.html#public-keys).
-
-Addresses have a length of 32 bytes.
+Celestia supports [secp256k1](https://en.bitcoin.it/wiki/Secp256k1) keys where [addresses](https://docs.cosmos.network/v0.46/basics/accounts.html#addresses) are 20 bytes in length. Addresses are prefixed with the [Bech32](https://en.bitcoin.it/wiki/Bech32) prefix `celestia`. For example, a valid address is `celestia1kj39jkzqlr073t42am9d8pd40tgudc3e2kj9yf`.
```
Comment (adlerjohn) [inline specs/src/specs/data_structures.md:114]: The HRP of the bech32 encoding (and indeed the encoding itself) is orthogonal to the actual address value. I think the specs should have a section for the value, and a section for the human-readable encoding used by frontends.

File: specs/src/specs/data_structures.md [hunk 3]
```diff
@@ -172,24 +167,18 @@ Unless otherwise indicated explicitly, objects are first [serialized](#serializa
 
 Consensus-critical data is authenticated using [ECDSA](https://www.secg.org/sec1-v2.pdf), with the curve [secp256k1](https://en.bitcoin.it/wiki/Secp256k1). A highly-optimized library is available in C (<https://github.com/bitcoin-core/secp256k1>), with wrappers in Go (<https://pkg.go.dev/github.com/ethereum/go-ethereum/crypto/secp256k1>) and Rust (<https://docs.rs/crate/secp256k1>).
 
-[Public keys](https://docs.cosmos.network/v0.46/basics/accounts.html#public-keys) are encoded in uncompressed form, as the concatenation of the `x` and `y` values. No prefix is needed to distinguish between encoding schemes as this is the only encoding supported.
+[Public keys](https://docs.cosmos.network/v0.46/basics/accounts.html#public-keys) are serialized in a compressed format.
```
Comment (adlerjohn) [inline specs/src/specs/data_structures.md:170]:
```suggestion
Public keys are serialized in a compressed format described [here](https://docs.cosmos.network/v0.46/basics/accounts.html#public-keys).
```

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (adlerjohn) [reply specs/src/specs/data_structures.md:174]: Are signatures _encoded_ as described here, or encoded with protobuf? Also, there's no `v` here?

Comment (rootulp) [reply specs/src/specs/data_structures.md:174]: I don't think `v` is supposed to be here. Ref: https://github.com/celestiaorg/celestia-app/issues/1440#issuecomment-1486806399

Comment (rootulp) [reply specs/src/specs/data_structures.md:174]: > Are signatures encoded as described here, or encoded with protobuf?

Good point. They appear to be encoded using protobuf per

https://github.com/celestiaorg/cosmos-sdk/blob/f19e20459b8bcf195167ad256de5a6ca6edc41f5/x/auth/tx/encoder.go#L13-L29

File: specs/src/specs/data_structures.md (hunk 2 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md:114]: I can extract a new section for the human-readable encoding but it seems like it doesn't belong on this page (data_structures.md). Proposal to extract a new page for `Public-Key Cryptography` which contains the section from this page along with a new section for human-readable Address encoding.

Comment (rootulp) [reply specs/src/specs/data_structures.md]: Done in https://github.com/celestiaorg/celestia-app/pull/1553/commits/cb52073e9cd3f9f69ef432d9d408db97c7767046

File: specs/src/specs/data_structures.md [hunk 4]
```diff
@@ -112,11 +111,11 @@ Abstraction over transaction fees.
...
 
-Address is a [type alias](#type-aliases).
+Celestia supports [secp256k1](https://en.bitcoin.it/wiki/Secp256k1) keys where [addresses](https://docs.cosmos.network/v0.46/basics/accounts.html#addresses) are 20 bytes in length.
 
-Addresses are the [hash](#hashing) [digest](#hashdigest) of the [public key](https://docs.cosmos.network/v0.46/basics/accounts.html#public-keys).
-
-Addresses have a length of 32 bytes.
+| name         | type       | description                                                             |
+|--------------|------------|-------------------------------------------------------------------------|
+| `AccAddress` | `[20]byte` | AccAddress a wrapper around bytes meant to represent an account address |
```
Comment (rootulp) [inline specs/src/specs/data_structures.md:118]: [for reviewers] inspired by https://github.com/celestiaorg/cosmos-sdk/blob/f19e20459b8bcf195167ad256de5a6ca6edc41f5/types/address.go#L130-L132

File: specs/src/specs/public_key_cryptography.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Public-Key Cryptography
+
+<!-- toc -->
+
+Consensus-critical data is authenticated using [ECDSA](https://www.secg.org/sec1-v2.pdf), with the curve [secp256k1](https://en.bitcoin.it/wiki/Secp256k1). A highly-optimized library is available in C (<https://github.com/bitcoin-core/secp256k1>), with wrappers in Go (<https://pkg.go.dev/github.com/ethereum/go-ethereum/crypto/secp256k1>) and Rust (<https://docs.rs/crate/secp256k1>).
```
Comment (liamsi) [inline specs/src/specs/public_key_cryptography.md:5]: Just noting that this file and section makes it sounds as if ecdsa/secp was the only pub-key crypto we were using. Validators still use ed25519 during consensus (which is consensus critical data).

Comment (adlerjohn) [reply specs/src/specs/public_key_cryptography.md:5]: Both signatures schemes should be described then.

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (adlerjohn) [reply specs/src/specs/data_structures.md]: In that case I would distinguish between the _value_ and the _encoding_ of that value.

Comment (rootulp) [reply specs/src/specs/data_structures.md]: Agreed. I attempted to in https://github.com/celestiaorg/celestia-app/pull/1553/commits/0d691af1d3c42427dc73ef936b7e055ecfec0524 and https://github.com/celestiaorg/celestia-app/pull/1553/commits/cb52073e9cd3f9f69ef432d9d408db97c7767046

File: specs/src/specs/public_key_cryptography.md (hunk 5 above)
Comment (rootulp) [reply specs/src/specs/public_key_cryptography.md:5]: Agreed will add a section for ed25519 based on https://github.com/cometbft/cometbft/blob/be6f4272d4ebedf264e3ce29475ffd31ecadff47/spec/core/encoding.md?plain=1#L42-L66

Comment (rootulp) [reply specs/src/specs/public_key_cryptography.md]: Added a section in https://github.com/celestiaorg/celestia-app/pull/1553/commits/64164c41784b508da778bd630c2f3c3db1362695

File: specs/src/specs/public_key_cryptography.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Public-Key Cryptography
+
+<!-- toc -->
+
+Consensus-critical data is authenticated using [ECDSA](https://www.secg.org/sec1-v2.pdf) with the curves: [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) or [Ed25519](https://en.wikipedia.org/wiki/EdDSA#Ed25519).
+
+## Secp256k1
+
+The Secp256k1 key type is used by accounts that submit transactions to be included in Celestia.
```
Comment (cmwaters) [inline specs/src/specs/public_key_cryptography.md:9]: Doesn't the SDK support multiple key types? Not just Secp256k1

**Review by cmwaters — APPROVED** [review]
LGTM

File: specs/src/specs/public_key_cryptography.md (hunk 6 above)
Comment (rootulp) [reply specs/src/specs/public_key_cryptography.md:9]: Yes it looks like it does. I can't seem to create one via celestia-appd though:

```go
$ ./build/celestia-appd keys add test --algo ed25519
Error: provided algorithm "ed25519" is not supported
```

so still investigating.

Ref:
1. https://docs.cosmos.network/v0.46/run-node/keyring.html#adding-keys-to-the-keyring

Comment (rootulp) [reply specs/src/specs/public_key_cryptography.md:9]: Relevant: https://github.com/cosmos/cosmos-sdk/discussions/8543

Comment (rootulp) [reply specs/src/specs/public_key_cryptography.md:9]: Resolving for now b/c I don't think we can actually use ed25519 keys in Cosmos SDK.

**Review by evan-forbes — APPROVED** [review]
:+1:

**Pull request merged**